File: backend/app/schemas/review.py
# app/schemas/review.py

from pydantic import BaseModel, Field
from datetime import datetime
from typing import Optional

from sqlalchemy import Column, Integer, String, Text, ForeignKey, DateTime
from sqlalchemy.sql import func
from app.core.database import Base


class ReviewBase(BaseModel):
    movie_id: int
    author: str = Field(..., max_length=100)
    content: str


class ReviewCreate(BaseModel):
    movie_id: int
    author: str
    content: str


class ReviewResponse(BaseModel):
    id: int
    movie_id: int
    author: str
    content: str

    sentiment_score: int
    sentiment_source: str
    created_at: datetime

    # ⭐⭐⭐ Pydantic v2 핵심
    model_config = {"from_attributes": True}

File: frontend/src/types/review.ts
//==============================================
//리뷰 엔티티 단일 책임
//감성 분석 결과 포함
//FastAPI schemas/review.py와 1:1 대응
//사용하는 곳: reviewApi.ts, ReviewList.tsx, ReviewForm.tsxMovieDetail.tsx
//==============================================

export interface Review {
  id: number;
  movie_id: number;
  author: string;
  content: string;
  sentiment_score: number;   // 1~5
  sentiment_source: string; // huggingface | rule-based
  created_at: string;
}

export interface ReviewCreate {
  movie_id: number;
  author: string;
  content: string;
}
